src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] map;
+
+    public MyHashMap() {
+        map = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= LOAD_FACTOR * map.length) {
+            resize();
+        }
+        int index = generateIndex(key);
+        if (map[index] == null) {
+            map[index] = new Node<>(key, value, null);
+            size++;
+        }
+        if (index != 0) {
```

Comment:
why so?

Reply:
if index == 0 we will rewrite value, because on index 0 we have Node with key == null

Reply:
not only null key can be stored on index 0, but any other key whose hash gives 0
collision in bucket with index 0 just includes null key